com.unity.shadergraph/Editor/Drawing/Inspector/GraphDataPropertyDrawer.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Reflection;
+using UnityEditor.ShaderGraph;
+using UnityEditor.ShaderGraph.Drawing;
+using UnityEditor.ShaderGraph.Internal;
+using UnityEngine.UIElements;
+
+namespace Drawing.Inspector
+{
+    [SGPropertyDrawer(typeof(GraphData))]
+    public class GraphDataPropertyDrawer : IPropertyDrawer
+    {
+        public delegate void ChangeConcretePrecisionCallback(ConcretePrecision newValue);
+        public delegate void PostTargetSettingsChangedCallback();
+
+        private PostTargetSettingsChangedCallback m_postChangeTargetSettingsCallback;
+        private ChangeConcretePrecisionCallback m_postChangeConcretePrecisionCallback;
+
+        // Targets have persistent data that needs to be maintained between interactions
+        // Hence we are storing a reference to this particular property drawer to continue using it
+        // private TargetPropertyDrawer m_TargetPropertyDrawer = new TargetPropertyDrawer();
+
+        //Dictionary<Target, bool> m_TargetFoldouts = new Dictionary<Target, bool>();
+
+        public void GetPropertyData(
+            PostTargetSettingsChangedCallback postChangeValueCallback,
+            ChangeConcretePrecisionCallback changeConcretePrecisionCallback)
+        {
+            m_postChangeTargetSettingsCallback = postChangeValueCallback;
+            m_postChangeConcretePrecisionCallback = changeConcretePrecisionCallback;
+        }
+
+        internal VisualElement CreateGUI(GraphData graphData)
+        {
+            if (graphData == null)
+                throw new InvalidCastException(
+                    "Attempting to draw something that isn't of type GraphData with a GraphDataPropertyDrawer");
+
+            var propertySheet = new VisualElement() {name = "graphSettings"};
+
+            var enumPropertyDrawer = new EnumPropertyDrawer();
+            propertySheet.Add(enumPropertyDrawer.CreateGUI(
+                newValue => { m_postChangeConcretePrecisionCallback((ConcretePrecision) newValue); },
+                graphData.concretePrecision,
+                "Precision",
+                ConcretePrecision.Float,
+                out var propertyVisualElement));
+
+            /* #TODO: Inspector - Uncomment when Matt lands stacks into master
```

Comment:
I'd rather you keep this code on a branch until stacks lands. Committing commented code makes later review harder.